Sources/Sentry/SentryHttpTransport.m

```diff
@@ -414,6 +414,13 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
                 return;
             }
 
+            if (response.statusCode >= 400 && response.statusCode < 500 && response.statusCode != 429) {
+                SENTRY_LOG_DEBUG(@"Received 4xx response code: %li", (long)response.statusCode);
```

Comment:
happy to make this specific to `400` status codes if we don't want to go broader, but generally speaking `4xx` is "client screwed up and shouldn't try again"

Reply:
Happy to let y'all take it over the line. Feel free to close this PR or push to it: whatever makes your lives easier.